RELEASE_NOTES.txt

```diff
@@ -1,6 +1,177 @@
+* Fixed an assertion error on IgniteCache#get.
+* Fixed assertion during RotatedIdPartRecord read from WAL.
+* Fixed backward connection check with loopback address.
+* Fixed cache dump check on a cache with a node filter.
+* Fixed cluster failure on topology change when DNS service is unavailable.
+* Fixed deserializing cache entries for IndexQuery on server.
+* Fixed duplicated keys written to dump.
+* Fixed entries cleaning on node shutdown/deactivation for in-memory mode.
+* Fixed entry and checkpoint locks orders in DataStreamer and update TTL.
+* Fixed excessive and useless warnings of IpFinderCleaner.
+* Fixed expire time setting for thin client replication.
+* Fixed fields count serialization in TcpCommunication messages.
+* Fixed incremental snapshot restore fail if consistent ID is not a string.
+* Fixed integer overflow in page eviction trackers.
+* Fixed iterator for local IndexQuery.
+* Fixed lambda serialization issues in code deployment for Java 21.
+* Fixed local class deployments cache cleanup on reconnect.
+* Fixed log message when Ignite Statistics usage state changes.
+* Fixed logging of reduced cache entries for distributed long running operations.
+* Fixed long JVM pause detector.
+* Fixed node crashing if a snapshot restore was canceled due to network issues.
+* Fixed node failure on timeout objects intersection.
+* Fixed node failure when a runtime exception occurs on snapshot start stage.
+* Fixed node failure while checking security permission for activation during join.
+* Fixed node join failure when index creation is in progress and caches have the same deploymentId.
+* Fixed pending tree redundant cleanup on entries expiration.
+* Fixed redundant lambda object allocation in ClockPageReplacementFlags.
+* Fixed removal of required affinity assignment instance from affinity cache.
+* Fixed sending a stack trace to a thin client if task fails in map phase.
+* Fixed server node fail when node service filter class is unknown.
+* Fixed service descriptor when deployment failed.
+* Fixed starting square bracket in default log4j2 pattern.
+* Fixed warnings about closed socket.
+* Fixed wrong old index size in the defragmentation procedure.
+* Forbid a non-transactional IgniteCache#clear() inside a transaction.
+* Implemented custom user defined metrics.
+* Implemented transaction aware SQL and Scan queries.
+* Improved check of previous node on new incoming server connection.
+* Moved Ignite sources to Java 11.
+* Optimized UUID serialization for communication messages.
+* Optimized entries expiration for persistent caches.
+* Removed deprecated TcpCommunicationSpi shared memory port configuration.
+* Removed deprecated authorization methods from Security Context.
+* Updated Jetty dependency version to 9.4.55.
+* Updated Netty dependency version to 4.1.101.Final.
+* Updated PostgreSQL JDBC dependency version to 42.7.3.
+* Updated Spring dependency version to 5.3.39.
+* Updated commons-io dependency version to 2.17.0.
```

Comment:
Should there be back quotes ` here and below?

`commons-io`, `grpc-context`, `json-path` etc